ci/cron/src/Main.hs

```diff
@@ -325,43 +221,62 @@ fetch_gh_paginated url = do
                 (_, _, _, [url, rel]) -> (rel, url)
                 _ -> fail $ "Assumption violated: link header entry did not match regex.\nEntry: " <> l
 
-fetch_gh_versions :: IO ([GitHubVersion], GitHubVersion)
+data Versions = Versions { top :: String, all_versions :: Set.Set String, dropdown :: [String] }
```

Comment:
I would recommend to store `SemVer.Version` here to make it explicit and only parse once.

Reply:
It's actually that exact same reasoning that led me to this representation: after the point where the `Versions` values are constructed, we only ever use them as `String`s.

If you look at all the calls to `parse` (there's only three of them), they never touch something extracted from a `Versions`. However, we do use the components of `Versions` as `String`s quite a bit, either to print for the user or to construct file paths.

I'll have to check if this is still true after trying to sort the hidden ones, though.

Reply:
I generally try to only convert to `String`s at the edges and retain as much structure an types as possible. But I don’t feel strongly about this here so feel free to ignore this comment.

Reply:
@nickchapman-da pushed for that too, so I eventually admitted I may be wrong and tried it out. I kinda like how it turned out.